examples/document_loading/csv/index.ts

```diff
@@ -0,0 +1,91 @@
+import Instructor from "@/instructor"
+import OpenAI from "openai"
+import { z } from "zod"
+import { CSVLoader } from "langchain/document_loaders/fs/csv";
+
+// Create a CSV loader instance
+const loader = new CSVLoader("./examples/document_loading/csv/transactions.csv")
+
+// Load the CSV document
+const doc = await loader.load()
+
+// // Define the schema for a single transaction
+// const TransactionsSchema = z.object({
+//   transactions: z
+//     .array(
+//       z.object({
+//         ticker: z.string(),
+//         date: z.string(),
+//         type: z.string(),
+//         qty: z.number(),
+//       })
+//     )
+// })
+
+// Define the schema for a single transaction
+const TransactionsSchema = z.object({
+  ticker: z.string(),
+  date: z.string(),
+  type: z.string(),
+  qty: z.number(),
+})
+
+// Initialize OpenAI and Instructor client
+const oai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY ?? undefined,
+  organization: process.env.OPENAI_ORG_ID ?? undefined
+})
+
+const client = Instructor({
+  client: oai,
+  mode: "TOOLS"
+})
+
+let extraction = {}
+
+const extractTransactions = await client.chat.completions.create({
+  messages: [{ role: "user", content: JSON.stringify(doc) }],
+  model: "gpt-4",
+  response_model: {
+    schema: TransactionsSchema,
+    name: "ExtractedTransactions"
```

Comment:
I guess I simply don't believe that converting a CSV file to a JSON file is something that needs a language model. 
We might want to come up with a better example. Because... It would probably just be easier to load the string. 

Or ask a language model to write the code to make this conversion.

Reply:
Just trying to provide an example for this [convo](https://x.com/dimitrikennedy/status/1748439278521127167?s=20). Wanted to make it applicable to Josh's original request and simple enough so people could see how they could relatively easily load a doc and extract data from it for use in their RAG pipelines. 

Happy to put something else together if you all think the CSV isn't illustrative enough.